File: subly_ios/Subly/Subly/UI/PaymentMethods/PaymentMethodsUiState.swift
import Foundation

struct PaymentMethodWithUsage: Identifiable {
    let paymentMethod: PaymentMethod
    let subscriptionCount: Int
    var id: UUID { paymentMethod.id }
}

struct PaymentMethodsUiState {
    var paymentMethods: [PaymentMethodWithUsage] = []
    var isLoading: Bool = true
    var error: String? = nil
    var showDeleteConfirmation: Bool = false
    var deletingMethod: PaymentMethodWithUsage? = nil
    var isDeleting: Bool = false
}

File: subly_android/app/src/main/java/net/cynreub/subly/ui/payment/PaymentMethodsUiState.kt
package net.cynreub.subly.ui.payment

import net.cynreub.subly.domain.model.PaymentMethod

data class PaymentMethodsUiState(
    val paymentMethods: List<PaymentMethodWithUsage> = emptyList(),
    val isLoading: Boolean = true,
    val error: String? = null,
    val showDeleteDialog: Boolean = false,
    val deletingPaymentMethod: PaymentMethodWithUsage? = null,
    val isDeleting: Boolean = false
)

data class PaymentMethodWithUsage(
    val paymentMethod: PaymentMethod,
    val subscriptionCount: Int
)
